security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -850,14 +829,14 @@ func (sc *SecretCache) getExchangedToken(ctx context.Context, k8sJwtToken string
 		cacheLog.Errorf("Found more than one plugin for %s", logPrefix)
 		return "", fmt.Errorf("found more than one plugin")
 	}
-	exchangedTokens, err := sc.sendRetriableRequest(ctx, nil, k8sJwtToken,
-		ConnKey{ConnectionID: "", ResourceName: ""}, false)
-	if err != nil || len(exchangedTokens) == 0 {
+	p := sc.configOptions.TokenExchangers[0]
```

Comment:
Shall we check if len(sc.configOptions.TokenExchangers) == 0, and return k8sJwtToken with an error message?

Reply:
We already do a few lines above (keeping existing logic):
```go
	if sc.configOptions.TokenExchangers == nil || len(sc.configOptions.TokenExchangers) == 0 {
		cacheLog.Debugf("Return k8s token for %s", logPrefix)
		return k8sJwtToken, nil
	}
```